### PR TITLE
Feature/delayed redirect

### PR DESCRIPTION
**Pull request by ae-mo**
Add a React component called `DelayedRedirect`, which enables us to redirect the user to a given path automatically after a specified amount of time has passed.

### CODE REVIEW

File: src/Router/Types.purs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+module React.Basic.Router.Types where
+
+import Data.Nullable (Nullable)
+import Foreign (Foreign)
+import React.Basic (JSX, ReactComponent)
```
Comment (kaol) [inline src/Router/Types.purs]: Unused import ReactComponent

File: src/Router/Foreign.purs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module React.Basic.Router.Foreign where
+
+import Data.Function.Uncurried (Fn0, runFn0)
+import Data.Nullable (Nullable)
```
Comment (kaol) [inline src/Router/Foreign.purs]: Unused

File: src/Router/Foreign.purs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import Foreign (Foreign)
```
Comment (kaol) [inline src/Router/Foreign.purs]: Unused

File: src/Router/Foreign.purs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import React.Basic (JSX, ReactComponent)
```
Comment (kaol) [inline src/Router/Foreign.purs]: JSX is unused

File: src/Router/DelayedRedirect.purs [hunk 5]
```diff
@@ -0,0 +1,37 @@
+module React.Basic.Router.DelayedRedirect where
+
+import Prelude
+
+import Effect (Effect)
```
Comment (kaol) [inline src/Router/DelayedRedirect.purs]: Unused

File: src/Router/DelayedRedirect.purs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import Effect.Class (liftEffect)
+import Effect.Class.Console as Console
```
Comment (kaol) [inline src/Router/DelayedRedirect.purs]: Unused

File: src/Router/DelayedRedirect.purs [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import Effect.Aff as Aff
+import React.Basic (JSX, element, make)
+import React.Basic as React
+import React.Basic.Router.Foreign as Router
+import React.Basic.Router.Types
```
Comment (kaol) [inline src/Router/DelayedRedirect.purs]: Unspecified imports, consider `(DelayedRedirectProps)`

File: src/Router/DelayedRedirect.purs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    didMount self@{ props: { delay }, setState, state } = do
+      Aff.launchAff_ do
+        Aff.delay $ Aff.Milliseconds delay
+        liftEffect $ setState _ { redirectState = Redirect }
+
+    render self@{ props: { from, push, to }, state: { redirectState } } =
+      case redirectState of
+        Wait     -> mempty
+        Redirect ->
+          let redirectProps = { from, push, to }
```
Comment (kaol) [inline src/Router/DelayedRedirect.purs:30]: It bothered me a bit to see `{ from, push, to }` repeated like this, but I couldn't quite see how to express it any better either. So I guess it's just ok.

**Review by kaol — CHANGES_REQUESTED** [review]
Spago gave me quite a few warnings, clean them up please.

File: src/Router/DelayedRedirect.purs (hunk 8 above)
Comment (ae-mo) [reply src/Router/DelayedRedirect.purs:30]: Yes, unfortunately this aspect is different from **Haskell**, AFAIK. But if we find a better way we can always change it later :smiley:

**Pull request merged**